src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -2042,10 +2042,10 @@ private void WriteRedFactoryWithNoAutoCreatableTokens(Node nd)
         private Field DetermineMinimalOptionalField(Node nd)
         {
             // first if there is a single list, then choose the list because it would not have been optional
-            int listCount = nd.Fields.Count(f => IsAnyNodeList(f.Type));
+            int listCount = nd.Fields.Count(f => IsAnyNodeList(f.Type) && !IsAttributeOrModifiersList(f));
```

Comment:
the changes in SourceWriter are so that we don't generate this syntax factory overload:

```
public ParenthesizedLambdaExpression ParethesizedLambda(SyntaxTokenList modifiers = default) { ... }
```

We already have 

```
public ParenthesizedLambdaExpression ParethesizedLambda() { ... }
```

Which this would conflict with.  Note: this change was already made in the local-function-attributes branch as well. this just cherry-picks that bac to here.

Reply:
I'm confused as to why there would be a `= default` optional value for the SyntaxTokenList when it is never optional. Is that just a special way that SyntaxTokenList gets handled in the generator? Sorry if this was already addressed when we pulled this into the other feature branch.

Reply:
> I'm confused as to why there would be a = default optional value for the SyntaxTokenList when it is never optional

So, first, Lists are *always* optional.  That's because they can be 'empty' and thus can be elided.

We then have extra logic in our source generator to say: let me generate a bunch of helper factory methods that should try to balance determining which nodes/tokens we will generate on their behalf, and which nodes they need to provide.

Let's look at an example of that:  BlockSyntax.

So a Block (prior to my attributes-on-statement work) was always: `{ statement* }`.

So, the generator saw that and said: ah, you have a single list, and some required tokens.  I will generate:

```public BlockSyntax Block(SyntaxList<StatementSyntax> statements = default)```

That's a really useful factory method.  You can trivially make an empty block with `Block()` and you can pass some amount of statements into that to generate a block out of those statements.

The logic for this code however, was that we would do this for the first list we found (as long as there was only one), otherwise we would use the first optional node.

When I added attributes to statements (in the other PR), and now that i've added 'modifiers' to lambdas (in this PR), the source generator says: ah, i've found a list that i think should be the one to focus these helper factory functions around.  For something like `Block` that would mean providing an entrypoint like:

```public BlockSyntax Block(SyntaxList<AttributeListSyntax> attributeLists = default)```

That's just literally not helpful.  No one is going to say "i want an easy way to make a block, optionally deciding on some attributes for it.  

In a very real sense, attributes and modifiers are *low* relevance.  They can be on a lot of nodes, but they're never the *primary* concern people have when constructing them simply.  So having the helper methods be targetted around them is *not* something we want.

Reply:
Makes total sense, thanks, and this seems like a reasonable enough one-off way of handling it instead of adding an extra attribute to the AttributeLists and Modifiers elements to express this case.